gitoxide-core/src/repository/log.rs

```diff
@@ -0,0 +1,152 @@
+use gix::bstr::{BStr, BString, ByteSlice};
+use gix::prelude::FindExt;
+use gix::ObjectId;
+
+pub fn log(mut repo: gix::Repository, out: &mut dyn std::io::Write, pathspec: BString) -> anyhow::Result<()> {
+    repo.object_cache_size_if_unset(repo.compute_object_cache_size_for_tree_diffs(&**repo.index_or_empty()?));
+
+    let head = repo.head()?.peel_to_commit_in_place()?;
+    let infos: Vec<_> =
+        gix::traverse::commit::topo::Builder::from_iters(&repo.objects, [head.id], None::<Vec<gix::ObjectId>>)
+            .build()?
+            .collect();
```

Comment:
Why collect here?
Generally collecting into a Vec seems unfeasible with any kind of commit-iteration.

Reply:
I was somehow under the impression that you had, in my `gix blame`-PR, made a similar change and introduced `.collect()`. But now I’m seeing that `blame_file` accepts `impl IntoIterator<…>`, so my assumption was incorrect and only based on the observation that callers of `blame_file` were using `.collect()`. I’ll remove the `.collect()`.